src/jit/morph.cpp

```diff
@@ -532,7 +532,7 @@ GenTree* Compiler::fgMorphCast(GenTree* tree)
     /* Just in case new side effects were introduced */
     tree->gtFlags |= (oper->gtFlags & GTF_ALL_EFFECT);
 
-    if (!gtIsActiveCSE_Candidate(tree)) // tree cannot be a CSE candidate
+    if (!gtIsActiveCSE_Candidate(tree) && !gtIsActiveCSE_Candidate(oper)) // tree cannot be a CSE candidate
```

Comment:
That comment should be removed, it was useless to begin with and now it's kind out of date as well.

But IMO it's not worth another CI build just for this.